CorsixTH/Lua/research_department.lua

```diff
@@ -343,8 +343,8 @@ function ResearchDepartment:addResearchPoints(points, autopsy_room)
   ---------------------- An autopsy has been done ---------------------------
   if autopsy_room then
     -- Do something only if the room is among those not yet discovered.
-    for room, _ in pairs(self.hospital.undiscovered_rooms) do
-      if room.id == autopsy_room then
+    for room_id, room in pairs(self.hospital.room_discovery) do
```

Comment:
Your `room` is not a room, it's a table containing a room and a discovery status. Don't use well-known names from the code if it is something different, it puts people on the wrong foot.
Perhaps `room_disc_status` ?

A trick I often employ is to use a plural word for a container, eg `hospital.room_discoveries`, which immediately gives you a trivial solution for the name of an element from that container (that is, `room_discovery` or shorter `room_disc`).

Reply:
Perhaps `room_availability` ?

Reply:
Yeah, this point is moot because the whole loop goes away and the condition below is just `if not self.hospital.room_discovery[autopsy_room].is_discovered then`

The logic in the loop depended on `room` which was different than the value lewri was using - so this is broken as is. Lewri, you need a way to get the room object from the room_id still.